src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Administration/GeneralDataProvider.php

```diff
@@ -0,0 +1,147 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\Administration;
+
+use Cookie;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShopBundle\Form\Exception\DataProviderError;
+use PrestaShopBundle\Form\Exception\DataProviderErrorCollection;
+use PrestaShopBundle\Form\Exception\DataProviderException;
+
+/**
+ * This class is responsible of managing the data manipulated using general form
+ * in "Configure > Advanced Parameters > Administration" page.
+ */
+final class GeneralDataProvider implements FormDataProviderInterface
+{
+    /**
+     * If you set cookie lifetime value too high there can be multiple problems.
+     * Hours are converted to seconds, so int might be turned to float if it's way to high.
+     * Cookie classes crashes if lifetime goes beyond year 9999, there are probably multiple other things.
+     * So we need to set some sort of max value. 100 years seems like a lifetime beyond reasonable use.
+     */
+    public const MAX_COOKIE_VALUE = 876000;
+
+    /**
+     * @var DataConfigurationInterface
+     */
+    private $dataConfiguration;
+
+    /**
+     * @var bool
+     */
+    private $sslEnabled;
+
+    /**
+     * @var bool
+     */
+    private $sslEnabledEverywhere;
+
+    public function __construct(
+        DataConfigurationInterface $dataConfiguration,
+        bool $sslEnabled,
+        bool $sslEnabledEverywhere
+    ) {
+        $this->dataConfiguration = $dataConfiguration;
+        $this->sslEnabled = $sslEnabled;
+        $this->sslEnabledEverywhere = $sslEnabledEverywhere;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getData()
+    {
+        return $this->dataConfiguration->getConfiguration();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setData(array $data)
+    {
+        $this->validate($data);
+
+        return $this->dataConfiguration->updateConfiguration($data);
+    }
+
+    /**
+     * Perform validations on form data.
+     *
+     * @param array $data
+     */
+    private function validate(array $data): void
+    {
+        $errors = new DataProviderErrorCollection();
+        $frontOfficeLifeTimeCookie = $data[GeneralType::FIELD_FRONT_COOKIE_LIFETIME];
+        $backOfficeLifeTimeCookie = $data[GeneralType::FIELD_BACK_COOKIE_LIFETIME];
```

Comment:
```suggestion
        $frontOfficeLifeTimeCookie = $data[GeneralType::FIELD_FRONT_COOKIE_LIFETIME];
        $backOfficeLifeTimeCookie = $data[GeneralType::FIELD_BACK_COOKIE_LIFETIME];
```

You assume data exists all the time? :thinking: 
A validator have to test if data are `isset`

Reply:
Good point. Thanks! Do you think it would be useful to throw error in that case? I don't see it happening elsewhere in PS and I don't see value of that in form. But if data provider to be used somewhere else it could be useful.

Reply:
Just an `isset` call should be enough :)

Reply:
Done